pkgs/by-name/pr/protonmail-desktop/package.nix

```diff
@@ -0,0 +1,95 @@
+{ lib, stdenv, fetchurl, wrapGAppsHook, makeWrapper, autoPatchelfHook, pkgs
+, alsa-lib
+, at-spi2-atk
+, at-spi2-core
+, atk
+, dpkg
+, glib
+, gtk3
+, libX11
+, libXtst
+, libdrm
+, mesa
+, nss
+, udev
+, xdg-utils
+, coreutils
+, xorg
+
+# command line arguments which are always set e.g "--disable-gpu"
+, commandLineArgs ? ""
+
+# Necessary for USB audio devices.
+, pulseSupport ? stdenv.isLinux
+, libpulseaudio
+
+# For GPU acceleration support on Wayland (without the lib it doesn't seem to work)
+, libGL
+
+# For video acceleration via VA-API (--enable-features=VaapiVideoDecoder,VaapiVideoEncoder)
+, libvaSupport ? stdenv.isLinux
+, libva
+}:
+
+let
+  inherit (lib) optional makeLibraryPath makeSearchPathOutput makeBinPath;
+
+  deps = [
+    alsa-lib at-spi2-atk at-spi2-core atk
+    glib gtk3 libdrm libX11 libGL
+    libXtst mesa nss udev xorg.libxcb
+  ]
+    ++ optional pulseSupport libpulseaudio
+    ++ optional libvaSupport libva;
+
+  rpath = makeLibraryPath deps + ":" + makeSearchPathOutput "lib" "lib64" deps;
+  binpath = makeBinPath deps;
+in
+stdenv.mkDerivation rec {
+  pname = "ProtonMail-desktop";
+  baseUrl = "https://proton.me/download/mail/linux/";
+  version = "beta";
```

Comment:
This sounds like they would just upload a new deb without updating the version, which would invalidate the hash.

Reply:
Can also be gotten from https://github.com/ProtonMail/inbox-desktop/releases, SHA256 matches exactly with `proton-mail_1.0.0_amd64.deb`

Reply:
Changed the url to github so it's versioned and guarded from this kind of modification by the provider.